lib/Runtime/Library/JavascriptBigInt.h

```diff
@@ -0,0 +1,71 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+namespace Js
+{
+    class JavascriptBigInt sealed : public RecyclableObject
+    {
+
+    private:
+        Field(uint32*) m_digits;         // digits
+        Field(uint32) m_length;          // length
+        Field(bool) m_isNegative;
+
+        // TODO: BigInt should be arbitrary-precision, need to resize space and add tagged BigInt
+        static const uint32 MaxStringLength = 100; // Max String length
+        static const uint32 MaxDigitLength = 20;  // Max Digit length 
+        Field(uint32) m_reservedDigitsInit[MaxDigitLength]; // pre-defined space to store array
+
+        DEFINE_VTABLE_CTOR(JavascriptBigInt, RecyclableObject);
+
+    public:
+        JavascriptBigInt(StaticType * type)
+            : RecyclableObject(type), m_length(0), m_digits(nullptr), m_isNegative(false)
+        {
+            Assert(type->GetTypeId() == TypeIds_BigInt);
+        }
+
+        JavascriptBigInt(const char16 * content, charcount_t cchUseLength, bool isNegative, StaticType * type)
+            : RecyclableObject(type)
```

Comment:
Could we call `JavascriptBigInt(type)` instead so all the fields get initialized?